File: 240430/작은 구슬의 이동/small-marble-movement.java
import java.util.Scanner;
public class Main {

    static int[] dy = new int[]{1,0,-1,0};
    static int[] dx = new int[]{0,-1,0,1};
    public static int dir(char c) {
        if(c=='R') // x좌표 증가
            return 0;
        if(c=='D') // y좌표 감소
            return 1;
        if(c=='L') //x좌표 감소
            return 2;
        if(c=='U') //y좌표 증가
            return 3;
        else
            return -1;
    }
    public static boolean isRange(int x, int y, int n) {
        return 1<x && x<n && 1<y && y<n;
    }

    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int t = sc.nextInt();

        int xpos = sc.nextInt();
        int ypos = sc.nextInt();
        char c = sc.next().charAt(0);
        int dir = dir(c);
        for(int i=0;i<t;i++) {
            if(xpos<1||xpos>n||ypos<1||ypos>n) {
                dir = (dir+2)%4;
                System.out.println("dir = " + dir);
                i--;
            }
            xpos += dx[dir];
            ypos += dy[dir];
            System.out.println(xpos+ " "+ ypos);
        }

        System.out.println(xpos+ " "+ ypos);
    }
}

File: 240424/작은 구슬의 이동/small-marble-movement.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    val dx = intArrayOf(1, 0, 0, -1)
    val dy = intArrayOf(0, 1, -1, 0)

    val n = sc.nextInt()
    var t = sc.nextInt()
    
    var r = sc.nextInt()
    var c = sc.nextInt()
    val d = sc.next()[0]

    var dirNum = getDir(d)

    while(t != 0) {
        val ry = r + dy[dirNum]
        val cx = c + dx[dirNum]

        if (inRange(n, ry, cx)) {
            r = ry
            c = cx
        } else {
            dirNum = 3 - dirNum
        }

        t--
    }

    print("$r $c")
}

fun inRange(n: Int, x: Int, y: Int): Boolean {
    return (x in 1..n) && (y in 1..n)
}

fun getDir(direction: Char): Int {
    return when (direction) {
        'R' -> { 0 }
        'D' -> { 1 }
        'U' -> { 2 }
        else -> { 3 }
    }
}